presto-main/src/main/java/io/prestosql/operator/scalar/UrlFunctions.java

```diff
@@ -128,15 +128,19 @@ public static Slice urlExtractParameter(@SqlType("varchar(x)") Slice url, @SqlTy
             return null;
         }
 
-        Slice query = slice(uri.getQuery());
         String parameter = parameterName.toStringUtf8();
-        Iterable<String> queryArgs = QUERY_SPLITTER.split(query.toStringUtf8());
+        Iterable<String> queryArgs = QUERY_SPLITTER.split(uri.getRawQuery());
```

Comment:
Change the call above to also use `uri.getRawQuery()`, for consistency and performance (it will do a decode)